test/SpokePoolClient.SpeedUp.ts

```diff
@@ -130,6 +130,9 @@ describe("SpokePoolClient: SpeedUp", function () {
     // Deposit is not returned until we reach deposit.quoteTimestamp.
     expect(spokePoolClient.getDepositsForDestinationChain(destinationChainId).length).to.equal(0);
     await spokePool.setCurrentTime(deposit.quoteTimestamp);
+
+    // Clear out spoke pool client data and re-update so that it should now see the formerly "early" deposit.
+    spokePoolClient = new SpokePoolClient(createSpyLogger().spyLogger, spokePool, null, originChainId, deploymentBlock);
```

Comment:
@pxrl this fixes the test because currently it does the following which I think is unrealistic in prod:
- updates with an early deposit. Chooses not to store the early deposit in memory
- updates with a speed up. There is no early deposit in memory so doesn't apply the speedup
- updates so that the deposit is no longer early. However, since its the still the same spoke pool client, it doesn't re query older blocks to look for deposit

Is it not more realistic to use a new spoke pool client before step 3?

Reply:
What if we just iterate over `this.speedUps` instead of `newSpeedUps` ? In the worst case we might end up hanging onto some irrelevant deposits, but this isn't an issue in serverless mode. It does however seem a bit more compatible with polling mode (which would otherwise be affected by this).

Reply:
https://github.com/across-protocol/sdk-v2/pull/455/commits/37c406b25e120fa4901867ae965f553b780befb6

So I tried this originally the only issue is that it creates a very ugly but not performance-affecting triple loop, because `this.speedUps` is a three-level deep object

Reply:
https://github.com/across-protocol/sdk-v2/pull/455/files#r1383630447